### ISSUE TITLE
持续学习 finetune loss

### ISSUE THREAD

**Issue by zizhec** [OpenBMB/MiniCPM#40]
看技术博客，wsd lrs 的效果表明，持续学习终究应该使用逐渐变小的loss，这意味着我们要finetune也得基于cpm最后的lr 开始下降。 
那cpm最后的lr是多少？

**Comment by ShengdingHu** [comment]
sft版本的最后学习率是1e-4 （base： 1e-2 -> 1e-3, sft： 1e-3 -> 1e-4) 
但是请注意这个学习率是结合了mup方案的。对于网络中的二维张量，以上学习率都要除1/9。因此，如果您不采用mup方案继续训，建议整体使用1e-5 左右的lr微调；如果您采用mup方案（https://arxiv.org/abs/2203.03466），则除了二维张量以外的参数使用1e-4，二维张量用1/9e-4。

Btw：我们dpo版本就是用了学习率是1e-5的cosine，并且没用mup方案。但是我们没有从dpo后的开始finetune的经验，不知道是从1e-5开始更好还是1e-6，我个人的猜测是如果想继续保持dpo模型的特性，可能1e-6更好，但是如果想做幅度更大的更新，1e-5更好

The final learning rate for the SFT version is 1e-4 (base: from 1e-2 to 1e-3, SFT from 1e-3 to 1e-4). However, please note that this learning rate is in conjunction with the mup scheme. For the two-dimensional tensors in the network, the above learning rates should be divided by 1/9. Therefore, if you do not adopt the mup scheme and continue training, it is recommended to fine-tune using an overall learning rate of about 1e-5. If you adopt the mup scheme ([link](https://arxiv.org/abs/2203.03466)), use 1e-4 for parameters other than two-dimensional tensors, and 1/9e-4 for two-dimensional tensors. 
BTW：In our DPO version, we used a learning rate of 1e-5 with cosine, and we did not use the mup scheme. However, we lack experience in fine-tuning from the point after DPO, so we are uncertain whether it is better to start with a learning rate of 1e-5 or 1e-6. My personal guess is that if you want to maintain the characteristics of the DPO model, a learning rate of 1e-6 might be better. However, if you aim for more significant updates, 1e-5 would be more appropriate.

**Comment by zizhec** [comment]
了解，多谢。

**Comment by ftgreat** [comment]
> sft版本的最后学习率是1e-4 （base： 1e-2 -> 1e-3, sft： 1e-3 -> 1e-4) 但是请注意这个学习率是结合了mup方案的。对于网络中的二维张量，以上学习率都要除1/9。因此，如果您不采用mup方案继续训，建议整体使用1e-5 左右的lr微调；如果您采用 [mup方案](https://arxiv.org/abs/2203.03466)，则除了二维张量以外的参数使用1e-4，二维张量用1/9e-4。
> 
> Btw：我们dpo版本就是用了学习率是1e-5的cosine，并且没用mup方案。但是我们没有从dpo后的开始finetune的经验，不知道是从1e-5开始更好还是1e-6，我个人的猜测是如果想继续保持dpo模型的特性，可能1e-6更好，但是如果想做幅度更大的更新，1e-5更好
> 
> The final learning rate for the SFT version is 1e-4 (base: from 1e-2 to 1e-3, SFT from 1e-3 to 1e-4). However, please note that this learning rate is in conjunction with the mup scheme. For the two-dimensional tensors in the network, the above learning rates should be divided by 1/9. Therefore, if you do not adopt the mup scheme and continue training, it is recommended to fine-tune using an overall learning rate of about 1e-5. If you adopt the mup scheme ([link](https://arxiv.org/abs/2203.03466)), use 1e-4 for parameters other than two-dimensional tensors, and 1/9e-4 for two-dimensional tensors. BTW：In our DPO version, we used a learning rate of 1e-5 with cosine, and we did not use the mup scheme. However, we lack experience in fine-tuning from the point after DPO, so we are uncertain whether it is better to start with a learning rate of 1e-5 or 1e-6. My personal guess is that if you want to maintain the characteristics of the DPO model, a learning rate of 1e-6 might be better. However, if you aim for more significant updates, 1e-5 would be more appropriate.

请问这里“二维张量”具体指哪些？谢谢
